test/simulator/metrics-collector-simulator.py

```diff
@@ -0,0 +1,355 @@
+            "name": "{}-clusters-metrics-collector-view".format(clusterName),
+            "annotations": {
+                "owner": "multicluster-operator",
+            },
+        },
+        "subjects": [{
+            "kind": "ServiceAccount",
+            "name": "endpoint-observability-operator-sa",
+            "namespace": clusterName
+        }],
+        "roleRef": {
+            "apiGroup": "rbac.authorization.k8s.io",
+            "kind": "ClusterRole",
+            "name": "cluster-monitoring-view",
+        },
+    }
+
+    try:
+        clusterRoleBindingAPI.create(body=metricsCollectorView)
+    except ApiException as err:
+        if err.status != 409:
+            return err
+
+
+def resetObject(obj: dynamic.ResourceInstance,
+                ns: str) -> dynamic.ResourceInstance:
+    obj.metadata.namespace = ns
+    obj.metadata.resourceVersion = None
+    obj.metadata.uid = None
+    obj.metadata.creationTimestamp = None
+    obj.metadata.managedFields = None
+
+    return obj
+
+
+def runSimulaterAt(hubClient: dynamic.DynamicClient,
+                   clusterName: str) -> ApiException:
+    if reuseAddonCertConfigMap(hubClient, clusterName) != None:
+        return ApiException(reason="failed to create cert configmap")
+
+    if reuseAddonSingerCertSecrets(hubClient, clusterName) != None:
+        return ApiException(reason="failed to create cert secrets")
+
+    if reuseAddonManagedCertSecrets(hubClient, clusterName) != None:
+        return ApiException(reason="failed to create cert secrets")
+
+    if reuseAddonCertServiceAccount(hubClient, clusterName) != None:
+        return ApiException(reason="failed to create cert serviceAccount")
+
+    if reuseAddonDeployment(hubClient, clusterName) != None:
+        return ApiException(reason="failed to create simulator deployment")
+
+    if createClusterRoleBinding(hubClient, clusterName) != None:
+        return ApiException(reason="failed to create clusterrolebinding")
+
+    return None
+
+
+def removeSimulator(hubClient: dynamic.DynamicClient,
+                    clusterNamePrefix: str) -> ApiException:
+    groupVersion = "cluster.open-cluster-management.io/v1"
+    kind = "ManagedCluster"
+    managedClusterAPI = hubClient.resources.get(api_version=groupVersion,
+                                                kind=kind)
+
+    clusterList = managedClusterAPI.get()
+
+    for item in clusterList.items:
+        clusterName = item.metadata.name
+        if clusterName.startswith(clusterNamePrefix):
+            try:
+                clusterRoleBindingAPI = hubClient.resources.get(
+                    api_version="rbac.authorization.k8s.io/v1",
+                    kind="ClusterRoleBinding")
+
+                clusterRoleBindingAPI.delete(
+                    name="{}-clusters-metrics-collector-view".format(
+                        clusterName))
+            except ApiException as err:
+                print("a", err.status)
+                if err.status != 404:
```

Comment:
if run into error, should we just log the error and continue to next cluster, but not return error?

Reply:
yup, you are right, let it fall through makes more sense when deleting